Operators as provided methods on Observable instead of extension traits (#78)
rxRust/rxRust: guide/developing.md

> ## To implement a operator tip?

Reviewer: ```suggestion
## How to implement an operator type
```

- Add a method in `Observable` to construct your operator type.
- Implement `Observable` for your operator type.

> - implement `LocalObservable` for your operator type to subscribe upstream and not care thread safe.

Reviewer: ```suggestion
- Implement `LocalObservable` for your operator type in order to handle a subscribe (upstream) for single-threaded use
```

> - implement `SharedObservable` for your operator type to subscribe upstream and keep thread safe.

Reviewer: ```suggestion
- Implement `SharedObservable` for your operator type in order to handle subscription (upstream) for multi-threaded use
```

## How to implement a creation operator?

> Creation operators are functions that can be used to create an Observable with some common predefined behavior. `rxRust` already have a generic struct named ObservableBase<LocalEmitter> for implement creation operators.

Reviewer: ```suggestion
Creation operators are functions that can be used to create an Observable with some common predefined behavior. `rxRust` already has a generic struct named `ObservableBase<LocalEmitter>` for implementing creation operators.
```

> ObservableBase<LocalEmitter> assume those Observables are accept a `Subscriber`  to emit values. So a easy way to implement a creation operator is :

Reviewer: ```suggestion
`ObservableBase<LocalEmitter>` assumes those Observables accept a `Subscriber`  to emit values. So an easy way to implement a creation operator is :
```

> - first, implement a `XxxEmitter` type which implement a `LocalEmitter` trait.

Reviewer: ```suggestion
- First, implement an `XxxEmitter` type which implements a `LocalEmitter` trait.
```

> - second, implement the function you want, this function just return a concrete type `ObservableBase<XxxEmitter>`.

Reviewer: ```suggestion
- Second, implement the function you want. It should return a concrete type `ObservableBase<XxxEmitter>`.
```

> - third, like `LocalObservable`, `LocalEmitter` has thread safe version named `SharedEmitter`, remember implement it for your `XxxEmitter`.

Reviewer: ```suggestion
- Third, like `LocalObservable`, `LocalEmitter` has a thread-safe version named `SharedEmitter`, remember to implement it for your `XxxEmitter`.
```
